src/Uno.UI/UI/Xaml/Controls/Control/Control.cs

```diff
@@ -208,9 +208,15 @@ public ControlTemplate Template
 			set { SetValue(TemplateProperty, value); }
 		}
 
-		// Using a DependencyProperty as the backing store for Template.  This enables animation, styling, binding, etc...
 		public static DependencyProperty TemplateProperty { get; } =
-			DependencyProperty.Register("Template", typeof(ControlTemplate), typeof(Control), new FrameworkPropertyMetadata(null, FrameworkPropertyMetadataOptions.ValueDoesNotInheritDataContext, (s, e) => ((Control)s)?.OnTemplateChanged(e)));
+			DependencyProperty.Register(
+				nameof(Template), 
+				typeof(ControlTemplate), 
+				typeof(Control), 
+				new FrameworkPropertyMetadata(
+					null, 
+					FrameworkPropertyMetadataOptions.ValueDoesNotInheritDataContext | FrameworkPropertyMetadataOptions.AffectsMeasure,
```

Comment:
I think that we should not change at large options like this ... I suspect that there are lot of case where the `InvalidateMeasure` was done in the property change callback. For instance this case you can see that the `INvalidateArrnge` is invoked here https://github.com/unoplatform/uno/pull/14253/files#diff-7e87918895f8b453fa65dab2ed76539d91c6f5c9f896ef2962aaadd7af2e121aR243 after some check, and noticeably a feature flag for that ...

Reply:
Agreed, in such case I would still keep the flag there, commented out with an explainer why it is deliberately missing in our case, because it will make it easier to understand when we make changes in the future